gocron.go

```diff
@@ -96,8 +97,8 @@ func (j *Job) run() (result []reflect.Value, err error) {
 	for k, param := range params {
 		in[k] = reflect.ValueOf(param)
 	}
-	result = f.Call(in)
-	j.lastRun = time.Now()
+	go f.Call(in)
+	j.lastRun = t
```

Comment:
how to ensure that all tasks finished when scheduler is going to be terminated, i'd better go for WaitGroup here instead of using `go`

Reply:
What benefit would a WaitGroup have?

Reply:
Is the job in your case here what would actually cancel the scheduler?  In that case a channel can be used that will schedule, and then the job itself could cancel if that is what is required.